pkg/kubectl/cmd/top_pod.go

```diff
@@ -111,11 +119,68 @@ func (o *TopPodOptions) Validate() error {
 	return nil
 }
 
-// RunTop implements all the necessary functionality for top.
 func (o TopPodOptions) RunTopPod() error {
-	metrics, err := o.Client.GetPodMetrics(o.Namespace, o.ResourceName, o.AllNamespaces, o.Selector)
+	var err error
+	selector := labels.Everything()
+	if len(o.Selector) > 0 {
+		selector, err = labels.Parse(o.Selector)
+		if err != nil {
+			return err
+		}
+	}
+	metrics, err := o.Client.GetPodMetrics(o.Namespace, o.ResourceName, o.AllNamespaces, selector)
+	// TODO: Refactor this once Heapster becomes the API server.
+	// First we check why no metrics have been received.
+	if len(metrics) == 0 {
+		// If the API server query is successful but all the pods are newly created,
+		// the metrics are probably not ready yet, so we return the error here in the first place.
+		e := verifyEmptyMetrics(o, selector)
+		if e != nil {
+			return e
+		}
+	}
 	if err != nil {
 		return err
 	}
 	return o.Printer.PrintPodMetrics(metrics, o.PrintContainers, o.AllNamespaces)
 }
+
+func verifyEmptyMetrics(o TopPodOptions, selector labels.Selector) error {
```

Comment:
Yes, this is related to Heapster's 'delay' in computing the metrics and not with a different API, I believe.
So even after the update, still for the first ~80s since pod creation there will be no metrics available, but we want to indicate that their creation is in progress (or that it failed in case the pod is not considered a newly created) - this is what I understood @vishh expected in #30826.